src/coreclr/src/utilcode/CMakeLists.txt

```diff
@@ -119,6 +126,14 @@ endif(CLR_CMAKE_HOST_WIN32)
 set_target_properties(utilcode_dac PROPERTIES DAC_COMPONENT TRUE)
 set_target_properties(utilcode_crossgen PROPERTIES CROSSGEN_COMPONENT TRUE)
 target_compile_definitions(utilcode_dac PRIVATE SELF_NO_HOST)
+target_compile_definitions(utilcode_dbi PRIVATE FEATURE_METADATA_EMIT)
```

Comment:
Where does utilcode depend on any of this?

Reply:
I'm not completely sure, but it is including the `src/coreclr/src/md/inc/metamodel.h` header.

Reply:
I can look into it more carefully.

Reply:
I guess it is including it to use it in pointer definitions. Those can be just forward decls e.g. `class CMiniMdSchema;`, without including the full header.

Reply:
I am guessing all these need to be fixed

```
git grep "#include .*\.\." -- src/coreclr/src/inc

src/coreclr/src/inc/eventtracebase.h:#include "../gc/env/etmdummy.h"
src/coreclr/src/inc/formattype.h:#include "../ildasm/dynamicarray.h"
src/coreclr/src/inc/metadata.h:#include "../md/inc/metamodelro.h"
src/coreclr/src/inc/metadata.h:#include "../md/inc/liteweightstgdb.h"
src/coreclr/src/inc/metadata.h:#include "../md/inc/metamodel.h"
src/coreclr/src/inc/stgpool.h:#include "../md/hotdata/hotheap.h"
src/coreclr/src/inc/stgpool.h:#include "../md/debug_metadata.h"
```

Reply:
Yes, it would be nice to not have dependencies from inc to other directories.

Reply:
I have created a separate change for part of this.  I will create issues to discuss the others.